engine/common/follower/engine.go

```diff
@@ -117,122 +141,149 @@ func New(
 	}
 	e.con = con
 
+	e.ComponentManager = component.NewComponentManagerBuilder().
+		AddWorker(e.processBlocksLoop).
+		Build()
+
 	return e, nil
 }
 
-// Ready returns a ready channel that is closed once the engine has fully
-// started. For consensus engine, this is true once the underlying consensus
-// algorithm has started.
+// Start starts follower loop, then the follower engine worker threads.
+func (e *Engine) Start(signalerContext irrecoverable.SignalerContext) {
+	e.log.Info().Msg("starting follower loop")
+	e.follower.Start(signalerContext)
+	e.log.Info().Msg("follower loop started")
+	e.ComponentManager.Start(signalerContext)
+	e.log.Info().Msg("follower engine started")
+}
+
+// Ready returns a ready channel that is closed once the engine has fully started.
+// For the follower engine, we wait for follower logic to start.
 func (e *Engine) Ready() <-chan struct{} {
-	return e.unit.Ready(func() {
-		<-e.follower.Ready()
-	})
+	// NOTE: this will create long-lived goroutines each time Ready is called
+	// Since Ready is called infrequently, that is OK. If the call frequency changes, change this code.
+	return util.AllReady(e.ComponentManager, e.follower)
 }
 
 // Done returns a done channel that is closed once the engine has fully stopped.
-// For the consensus engine, we wait for hotstuff to finish.
+// For the follower engine, we wait for follower logic to finish.
 func (e *Engine) Done() <-chan struct{} {
-	return e.unit.Done(func() {
-		<-e.follower.Done()
-	})
+	// NOTE: this will create long-lived goroutines each time Done is called
+	// Since Done is called infrequently, that is OK. If the call frequency changes, change this code.
+	return util.AllDone(e.ComponentManager, e.follower)
 }
 
-// SubmitLocal submits an event originating on the local node.
-func (e *Engine) SubmitLocal(event interface{}) {
-	e.unit.Launch(func() {
-		err := e.process(e.me.NodeID(), event)
-		if err != nil {
-			engine.LogError(e.log, err)
-		}
-	})
+// Process processes the given event from the node with the given origin ID in
+// a blocking manner. It returns the potential processing error when done.
+// TODO(active-pacemaker): As part of https://github.com/dapperlabs/flow-go/issues/6424 update process
+// to accept events.SyncedBlock only through trusted interface instead of general one.
+func (e *Engine) Process(channel channels.Channel, originID flow.Identifier, message interface{}) error {
+	switch msg := message.(type) {
+	case *events.SyncedBlock:
+		return e.onSyncedBlock(originID, msg)
+	case *messages.BlockProposal:
+		e.onBlockProposal(originID, msg)
+	case *messages.BlockResponse:
+		e.onBlockResponse(originID, msg)
```

Comment:
I think we can remove this. This was added to [solve a race condition](https://github.com/onflow/flow-go/pull/2721) which doesn't exist any more with Active Pacemaker, and the `sync` engine doesn't forward `BlockResponse` messages here any more.
```suggestion
```